spec/enumerable_spec.rb

```diff
@@ -0,0 +1,213 @@
+require_relative '../enumerable'
+
+describe Enumerable do
+  let(:arr) { [1, 2, 3, 4, 5] }
+  let(:hash) { { one: 'first', two: 'second', three: 'third', four: 'fourth', five: 'fifth' } }
+  let(:arr_nil) { [nil, 1, 2, 3, 4, 5] }
+  let(:arr_false) { [1, 2, 3, 4, 5, false] }
+  let(:arr_string) { %w[one two three four five] }
+  let(:arr_none) { [false, nil, false, nil, false] }
+
+  describe '#my_each' do
+    it 'returns an enumerator if no block given' do
+      expect(arr.my_each).to be_a(Enumerator)
+    end
+
+    it 'iterate over the array, and updates new array' do
+      new_variable = []
+      arr.my_each { |x| new_variable << x * 2 }
+      expect(new_variable).to eq([2, 4, 6, 8, 10])
+    end
+
+    it 'iterate over the array, but returns the same array' do
+      expect(arr.my_each { |x| x * 2 }).to eql arr
+    end
+  end
+
+  describe 'my_each_with_index' do
+    it 'calls block with two arguments, the item and its index, iterate over argument and update new variable' do
+      new_variable = []
+      arr.my_each_with_index { |x, _index| new_variable << x * 2 }
+      expect(new_variable).to eql([2, 4, 6, 8, 10])
+    end
+
+    it 'returns enumerator if no block given' do
+      expect(hash.my_each_with_index).to be_a(Enumerator)
+    end
+
+    it 'calls block with two arguments, the item and its index, iterate over argument and return same argument' do
+      expect(arr.my_each_with_index { |x, index| x * 2 if index.odd? }).to eql arr
+    end
+  end
+
+  describe '#my_select' do
+    it 'returns enumerator if no block given' do
+      expect(hash.my_select).to be_a(Enumerator)
+    end
+
+    it 'returns an array of elements for which the given block returns a true value' do
+      expect(arr.my_select { |x| x < 4 }).to eql([1, 2, 3])
+      expect(hash.my_select { |_k, v| %w[second fifth].include?(v) }).to be_a(Array)
+    end
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

Reply:
Finished!